src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas1/Cas1SpaceBookingController.kt

```diff
@@ -179,13 +197,29 @@ class Cas1SpaceBookingController(
   ): ResponseEntity<Unit> {
     userAccessService.ensureCurrentUserHasPermission(UserPermission.CAS1_SPACE_BOOKING_RECORD_DEPARTURE)
 
+    val departureDateAndTime = if (cas1NewDeparture.departureDateTime != null) {
+      Pair(
+        cas1NewDeparture.departureDateTime!!.toLocalDate(),
+        cas1NewDeparture.departureDateTime!!.toLocalDateTime().toLocalTime(),
+      )
+    } else {
+      val departureDate = cas1NewDeparture.departureDate
+        ?: throw BadRequestProblem(invalidParams = mapOf("departureDate" to "is required"))
+      val departureTime = cas1NewDeparture.departureTime
+        ?: throw BadRequestProblem(invalidParams = mapOf("departureTime" to "is required"))
+      Pair(
+        departureDate,
+        LocalTime.parse(departureTime),
+      )
+    }
```

Comment:
in fact as we are repeating could go with this:
```
val departureDateAndTime = getDateTime(
      dateTime = cas1NewDeparture.departureDateTime,
      date = cas1NewDeparture.departureDate,
      time = cas1NewDeparture.departureTime
    )
```
to call this:

```
private fun getDateTime(dateTime: Instant?, date: java.time.LocalDate?, time: String?) = if (dateTime != null) {
      Pair(
        first = dateTime.toLocalDate(),
        second = dateTime.toLocalDateTime().toLocalTime(),
      )
    } else {
      time ?: throw BadRequestProblem(invalidParams = mapOf("departureTime" to "is required"))
      Pair(
        first = date ?: throw BadRequestProblem(invalidParams = mapOf("departureDate" to "is required")),
        second = LocalTime.parse(time),
      )
    }
```

Reply:
then above re-used for arrivalDateAndTime also - prob obvious that bit :)

Reply:
Yep definitely looks better, but as temporary code probably not worth the refactor. Note that once `cas1NewDeparture.departureDateTime` is removed (it's deprecated), we'll make the other two fields mandatory in the API so we won't need to do these checks here either.

(sorry, i'm using temporary code as an excuse for not doing anything :) )

Reply:
haha, that's fine - temp code as you say :)